class.jetpack.php

```diff
@@ -577,23 +585,38 @@ static function delete_active_modules() {
 	public function add_configure_hook() {
 		global $wp_filter;
 
-		$current_priority = has_filter( 'plugins_loaded', array( $this, 'configure' ) );
-		if ( false !== $current_priority ) {
-			remove_action( 'plugins_loaded', array( $this, 'configure' ), $current_priority );
+		// If there is no handler set for the plugins_loaded hook yet, set it and return.
+		if ( ! isset( $wp_filter['plugins_loaded'] ) ) {
+			add_action( 'plugins_loaded', array( $this, 'configure' ), 0 );
+			$this->configure_hook_priority = 0;
+			return;
 		}
 
-		$taken_priorities = array_map( 'intval', array_keys( $wp_filter['plugins_loaded']->callbacks ) );
-		sort( $taken_priorities );
+		$highest_priority = min( array_keys( $wp_filter['plugins_loaded']->callbacks ) );
 
-		$first_priority = array_shift( $taken_priorities );
+		// If our hook is set, and it's still at highest priority, we don't need to do anything.
+		if (
+			isset( $this->configure_hook_priority )
 
-		if ( defined( 'PHP_INT_MAX' ) && $first_priority <= - PHP_INT_MAX ) {
-			$new_priority = - PHP_INT_MAX;
-		} else {
-			$new_priority = $first_priority - 1;
+			// Priority is in reverse order, so less is higher.
+			&& $this->configure_hook_priority <= $highest_priority
+		) {
+			return;
```

Comment:
A unit test failed because `Jetpack::configure_hook_priority` had a value but the configure hook on plugins_loaded didn’t actually exist. I’m not sure if it’s possible to get into this state in production. To be on the safe side, what do you think of checking that the hook exists and adding it if it doesn’t before returning here?

Reply:
Done in the latest commit, thanks!